src/macports1.0/sysctl.c

```diff
@@ -92,7 +91,36 @@ int SysctlCmd(ClientData clientData UNUSED, Tcl_Interp *interp, int objc UNUSED,
     Tcl_SetObjResult(interp, tcl_result);
     return TCL_OK;
 #else
-    Tcl_SetObjResult(interp, Tcl_NewStringObj("sysctl not available", -1));
-    return TCL_ERROR;
+    char *name;
+    name = Tcl_GetString(objv[1]);
+    if (objc != 2) {
+        Tcl_WrongNumArgs(interp, 1, objv, "name");
+        return TCL_ERROR;
+    }
+
+    if (strcmp(name, "hw.activecpu") == 0) {
+        int res = sysconf(_SC_NPROCESSORS_ONLN);
+        if (res < 1)
+        {
+            Tcl_SetObjResult(interp, Tcl_NewStringObj("sysconf not available", -1));
+            return TCL_ERROR;
+        }
+        Tcl_SetObjResult(interp, Tcl_NewIntObj(res));
+        return TCL_OK;
+    } else if (strcmp(name, "hw.memsize") == 0) {
+        long pages = sysconf(_SC_PHYS_PAGES);
+        long page_size = sysconf(_SC_PAGE_SIZE);
+        unsigned long long res = pages * page_size;
+        if (res < 1)
```

Comment:
Wouldn't this incorrectly return `1` if both calls to `sysconf` fail?

Why not
```suggestion
        if (pages < 0 || page_size < 0)
```
or with 1, although strictly speaking that's not documented as an error in the manpage.